### DISCUSSION TITLE
How to get the unit in it's abbreviated form (a.k.a get `m` from unit.toJSON().unit instead of `meters`?

### DISCUSSION THREAD

**Discussion by rafaellehmkuhl** [josdejong/mathjs#3236] [Q&A]
I have a situation where I want to allow the user to choose the unit in which a variable can be displayed. The `Unit` module of `mathjs` is perfect for that.

Currently I can access `unit(x, 'm').toJSON().unit` and receive `meters`. 
Is there a way I can have this unit name in it's abbreviated form (`m`)?
```

**Comment by josdejong** [comment]
`unit(x, 'm')` does not replace `m` with `meter` by itself but that an happen after operations on the unit.

There is no built-in solution to ensure outputting abbreviations only. Maybe you can create list with the unit names and their abbreviations, and write a RegExp to replace them in the output (mind any prefixes like `kilometer` and `millimeter`). Or use `unit.to('m')` to convert to the abbreviation but I think that is harder to implement.
